vignettes/readr.Rmd

```diff
@@ -160,60 +151,38 @@ readr comes with five parsers for rectangular file formats:
 * `read_fwf()` for fixed-width files
 * `read_log()` for web log files
 
-Each of these functions firsts calls `spec_xxx()` (as described above), and then parses the file according to that column specification:
+If readr finds problems with parsing your data, it will emit a warning message.
+For example, in our `challenge.csv` file, the column type we expected from our data was a column of dates and a column of doubles.
 
 ```{r}
-df1 <- read_csv(readr_example("challenge.csv"))
+df1 <- read_csv(readr_example("challenge.csv"), col_types = list(
+  date = col_date(),
+  value = col_double()
+))
 ```
 
-The rectangular parsing functions almost always succeed; they'll only fail if the format is severely messed up. Instead, readr will generate a data frame of problems. The first few will be printed out, and you can access them all with `problems()`:
+The data was successfully imported, but we get a warning message about parsing issues.
+To view more details about these parsing issues, call `problems()` on your data frame.
+This will generate a data frame with information about where the problem was found, and why.
 
 ```{r}
 problems(df1)
 ```
 
-You've already seen one way of handling bad guesses: increasing the number of rows used to guess the type of each column. 
-
-```{r}
-df2 <- read_csv(readr_example("challenge.csv"), guess_max = 1001)
-```
-
-Another approach is to manually supply the column specification.
-
-### Overriding the defaults
-
-In the previous examples, you may have noticed that readr printed the column specification that it used to parse the file:
-
-```{r}
-#> Parsed with column specification:
-#> cols(
-#>   x = col_integer(),
-#>   y = col_character()
-#> )
-```
-
-You can also access it after the fact using `spec()`:
-
-```{r}
-spec(df1)
-spec(df2)
+In this example, one of the dates is in an incorrect format.
+Since `problems()` gives us the row number from the original file, fixing this value in the original file could be easy, especially since there is only one problematic value.
+But data in the wild is rarely so clean.
+We can fix this parsing error by importing the date column as character and using the lubridate package to parse our dates.
+
+```{r, eval = FALSE}
+df1 <- read_csv(readr_example("challenge.csv"), col_types = list(
+  date = col_character(),
+  value = col_double()
+))
+df1$date <- lubridate::ymd(df1$date)
```

Comment:
Even though we can't run this due to the lubridate call, I think it serves as a good example of a problem you'd see in the wild (ie mixed bag of date formats). And here is an example of such a problem:
https://github.com/tidyverse/readr/issues/737

Reply:
Yeah I'm fine with showing the use of lubridate. I think you should add some brief comment that we can't execute this chunk, since readr doesn't depend on lubridate. But that it's indicative of a general type of workaround, where you import as character and use a more specialized package (or custom function) to parse a tricky column.